### PR TITLE
Feature/tsmo use case two

### PR DESCRIPTION
**Pull request by aswath1**
# PR Details
## Description

The implementation closely followed that of stop_controlled_intersection_tactical_plugin and sci_strategic_plugin. This is tactical_plugin implementation. In the future there should be light_controlled_intersection_strategic_plugin (or lci_strategic_plugin),that gives this plugin when to enter the intersection through float_meta_data.

As CARMA Platform operating on a vehicle approaching a fixed-time traffic signal, I want to use the Signal Phase and Timing (SPaT) message to plan a trajectory such that the ego vehicle does not have to stop prior to the light turning green so that traffic flow behind the vehicle does not suffer a shockwave. Depending on SPaT, four main trajectory cases are possible. I should determine which trajectory case I should follow.

Acceptance Criteria:

Depending on SPaT (the time when the vehicle can depart from the intersection), the Trajectory Plan can follow one of the following four main cases:

CASE 1: This case will assume that the target departure time is soon enough based on the vehicle’s present location and speed, so that the vehicle should first accelerate, then cruise with the maximum speed (speed limit), and then decelerate to leave the intersection at the target speed. In this case, the trajectory profile include three sections: acceleration, cruising, and deceleration (with respect to the order). The cruising speed will be at speed limit.

CASE 2: This case will assume that the target departure time is soon enough (but not sooner than what is required for CASE 1) based on the vehicle’s present location and speed, so that the vehicle should first accelerate and then decelerate to leave the intersection at the target speed. In this case, the trajectory profile include two sections: acceleration and deceleration (with respect to the order).

CASE 3: This case will assume that the target departure time is late enough based on the vehicle’s present location and speed so that vehicle should first decelerate and then accelerate to leave the intersection at the target speed. In this case, the trajectory profile include two sections: deceleration and acceleration (with respect to the order).

CASE 4: This case will assume that the target departure time is in the distant future based on the vehicle’s present location and speed so that vehicle should first decelerate, then cruise with a minimum configurable speed (e.g., 5 mph), and then accelerate to leave the intersection at the target speed. In this case, the trajectory profile include three sections: deceleration, cruising, and acceleration (with respect to the order). The cruising speed will be a minimum configurable value.

To determine which case should the vehicle follow, please refer to the Trajectory Smoothing (TS) section of the algorithm design.

Note: the departure target speed can be set equal to the speed limit or a lower speed for safety concerns.

Note: the minimum cruising speed can be set as a configurable parameter, so that in Case 4, the vehicle can proceed with a minimum speed instead of stopping prior to the stop bar.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [x] I have added any new packages to the sonar-scanner.properties file
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: light_controlled_intersection_tactical_plugin/include/light_controlled_intersection_plugin_node.h
Comment (aswath1) [reply light_controlled_intersection_tactical_plugin/include/light_controlled_intersection_plugin_node.h:81]: Default values are ints, and the function it is being used also accept ints

File: light_controlled_intersection_tactical_plugin/src/light_controlled_intersection_tactical_plugin.cpp
Comment (aswath1) [reply light_controlled_intersection_tactical_plugin/src/light_controlled_intersection_tactical_plugin.cpp:219]: Yeah it's mistake. This plugin's one was intended to be used. The difference is that this function changes the maneuver type to LANE_FOLLOW from INTERSETION_TRANSIT to take advantage of existing create_lanefollow_geometry library function.  I thought adding another INTERSETION_TRANSIT option in the create_geometry_profile  library itself might break other plugin's that send that maneuver type. Please let me know how we can distinguish this and other plugin's there?

File: light_controlled_intersection_tactical_plugin/src/light_controlled_intersection_tactical_plugin.cpp
Comment (aswath1) [reply light_controlled_intersection_tactical_plugin/src/light_controlled_intersection_tactical_plugin.cpp]: this one intended to use library function. And it did not give error

File: light_controlled_intersection_tactical_plugin/src/light_controlled_intersection_tactical_plugin.cpp
Comment (msmcconnell) [reply light_controlled_intersection_tactical_plugin/src/light_controlled_intersection_tactical_plugin.cpp:219]: basic_autonomy could be far more generic than it is, but I don't think we have time for a rework of it at the moment. I think the solution you have here is a good work around.

**Comment by msmcconnell** [comment]
LGTM. I'll wait for CI to run for a bit before merging

**Review by msmcconnell — APPROVED** [review]

**Pull request merged**
